### ISSUE TITLE
Improve startup performance logging

### ISSUE THREAD

**Issue by sgraband** [eclipse-theia/theia#9964]
### Feature Description:
In addition to #9722 we would like to log some additional information regarding the startup process to easily monitor the startup process. As opposed to the script introduced by #9722 the logs would be measured from within the application to gain better understanding which processes increase the startup time. As noted in #9868 the plugin deployment mechanism seems to impact the startup time noticeably.

There are already measurements for the plugin mechanism logged (namely deploy, sync, load and start). However, only the duration of these steps is logged. In our opinion, it makes sense to track the time it takes to finish these steps from the frontend start. This way potential delays can be recognized earlier, as you also get information about when these processes are happening, as opposed to only knowing how long they take. This would also help to identify potential other influences on startup time.

On top of that, we noticed a problem with the measurement of the sync-step. If the deployment on the backend is not yet finished the sync cannot proceed. Hence you get long times for the sync block, when the deployment is the problem. This could easily be changed, by either starting the timer after the waiting or, preferably, by splitting the sync step in two separate steps: waiting for deployment and sync.

To summarize, we would suggest logging the timestamps from the frontend start in addition to the existing measurements, as well as splitting the sync step into two separate measurements. Additionally, we would also log important UI steps (loading indicator is shown, UI is loaded, etc.) to get an easy way of seeing the impact certain changes have. 

Please let us know how you feel about this and if you have any suggestions. Thanks in advance.
